samples/Directory.Build.props

```diff
@@ -33,4 +33,8 @@
     <MtouchUseLlvm>false</MtouchUseLlvm>
   </PropertyGroup>
 
+  <!-- AOT not supported for osx-arm64 on .NET 7 -->
+  <PropertyGroup Condition="'$(SolutionName)' == 'Sentry.Aot' and (!$([MSBuild]::IsOSPlatform('OSX')) or '$(TargetFramework)' == 'net8.0')">
+    <PublishAot>true</PublishAot>
+  </PropertyGroup>
```

Comment:
I still believe this should be done in each sample individually as needed. Not all samples support AOT publishing nor do we care since we don't normally run `dotnet publish` on them - we only do that in the CLI integration test for the basic console sample.
```suggestion
```